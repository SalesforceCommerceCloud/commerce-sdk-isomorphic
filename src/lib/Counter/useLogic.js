import {useState} from 'react';

const useLogic = initialState => {
  const [count, setCount] = useState(initialState);
  return {
    count,
    incrementCount: () => setCount(count + 1),
  };
};

export default useLogic;
