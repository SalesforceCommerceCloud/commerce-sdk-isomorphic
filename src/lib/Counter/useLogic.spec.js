import {renderHook, act} from '@testing-library/react-hooks';
import useLogic from './useLogic';

describe('lib > Counter > useLogic', () => {
  const initialValue = 0;
  it('renders', () => {
    /**
     * Render hook, using testing-library utility
     * @see https://react-hooks-testing-library.com/reference/api#renderhook
     */
    const {result} = renderHook(() => useLogic(initialValue));

    expect(result.current).toMatchSnapshot();
  });
  it('increments value', () => {
    /**
     * Render hook, using testing-library utility
     * @see https://react-hooks-testing-library.com/reference/api#renderhook
     */
    const {result} = renderHook(() => useLogic(initialValue));

    /**
     * Wrap state update with act
     * @see https://react-hooks-testing-library.com/usage/basic-hooks#updates
     * @see https://fb.me/react-wrap-tests-with-act
     */
    act(() => {
      result.current.incrementCount();
    });

    expect(result.current.count).toBe(initialValue + 1);
  });
});
