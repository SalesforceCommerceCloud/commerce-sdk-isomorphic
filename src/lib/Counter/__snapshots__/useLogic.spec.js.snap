// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib > Counter > useLogic renders 1`] = `
Object {
  "count": 0,
  "incrementCount": [Function],
}
`;
